assets/tests/components/routeLadder.test.tsx

```diff
@@ -259,6 +259,89 @@ describe("routeLadder", () => {
     expect(tree).toMatchSnapshot()
   })
 
+  test("renders descriptive content in dropdown if there's a non-skate alert", async () => {
+    jest.mocked(getTestGroups).mockReturnValue([TestGroups.DetoursPilot])
+
+    const route: Route = routeFactory.build({
+      id: "28",
+      name: "28",
+    })
+    const timepoints = [
+      { id: "MATPN", name: "MATPN Name" },
+      { id: "WELLH", name: "WELLH Name" },
+      { id: "MORTN", name: "MORTN Name" },
+    ]
+
+    render(
+      <RouteLadder
+        route={route}
+        timepoints={timepoints}
+        vehiclesAndGhosts={undefined}
+        selectedVehicleId={undefined}
+        deselectRoute={() => {}}
+        reverseLadder={() => {}}
+        toggleCrowding={() => {}}
+        ladderDirections={{}}
+        ladderCrowdingToggles={{}}
+        hasAlert={true}
+      />
+    )
+
+    act(() => {
+      fireEvent.click(screen.getByRole("button", { name: "28 Route Options" }))
+    })
+
+      expect(
+      await screen.findByText("This route has an active detour.")
```

Comment:
```suggestion
      screen.getByText(/This route has an active detour./)
```

The problem here isn't the waiting - `act(() => fireEvent...)` is running through the full round of react loops and making the dropdown visible, so that part is working just fine.

The problem is that `findByText` (and its friends `getByText`, `queryByText`, etc) do full-string matching, not substring matching, so if the text content here is `This route has an active detour. View detour details on mbta.com or IRIS.`, then `findByText` won't find it.

There are a few ways to tell `findByText` to use substring matching. I think replacing string `"This route has an active detour."` with regex `/This route has an active detour./` looks nicest, but you can also explicitly tell `findByText` to do substring matching by giving it the `exact: false` option:

```suggestion
      await screen.findByText("This route has an active detour.", { exact: false })
```

(In my original suggestion, I also replaced `findByText` with `getByText`, because the race condition isn't the problem here, and I would recommend doing that either way)

```suggestion
      screen.getByText("This route has an active detour.", { exact: false })
```

Reply:
I had no ideaaaaaa it was a string matching issue! Fantastic, thanks for help and breaking down my options